src/askui/models/shared/agent.py

```diff
@@ -165,6 +165,9 @@ def _use_tools(
             for content_block in message.content
             if content_block.type == "tool_use"
         ]
+        if len(tool_use_content_blocks) == 0:
+            return None
```

Comment:
WDYT about removing this? `ThreadPoolExecutor.map()` and, therefore, `ToolCollection.run()` can handle empty list.